tests/Api/Response/ResponseParserTest.php

```diff
@@ -73,16 +73,46 @@ public function testParse()
         ]);
     }
 
-    public function testHandleErrorResponse()
+    public function responseDataProvider()
     {
-        $errorResponse = [
+        $plentyErrorResponse = [
             'error' => true,
             'error_no' => 0,
             'error_msg' => 'Curl error: Could not resolve host: service.findologic.com',
             'error_file' => '/findologic/http_request2/HTTP/Request2/Adapter/Curl.php',
             'error_line' => 155
         ];
 
+        return [
+            'Plentymarkets error response' => [
+                'response' => $plentyErrorResponse,
+                'expectedLog' => [
+                    'Still invalid response after 2 retries. Using Plentymarkets SDK results without Findologic.',
+                    ['response' => $plentyErrorResponse],
+                ]
+            ],
+            'Empty response' => [
+                'response' => '',
+                'expectedLog' => [
+                    'Still invalid response after 2 retries. Using Plentymarkets SDK results without Findologic.',
+                    ['response' => ''],
+                ]
+            ],
+            'Invalid XML response' => [
+                'response' => 'invalid-xml',
+                'expectedLog' => [
+                    'Parsing XML failed',
+                    ['xmlString' => 'invalid-xml'],
+                ]
+            ],
+        ];
+    }
+
+    /**
+     * @dataProvider responseDataProvider
+     */
+    public function testHandleInvalidResponse($response, $expectedLog)
```

Comment:
Add some typehints to the arguments. It is also not clear what type `$expectedLog` is, by the name. Maybe rename it to `$expectedLogData`, so it is more clear what it contains.

```suggestion
    /**
     * @param string|array $response
     */
    public function testHandleInvalidResponse($response, array $expectedLog)
```

Reply:
Maybe also split up `$errorMessage` and `$context` into separate variables. This way you don't have to access the array indexes in line 127.